packages/api/src/command/medical/patient/converage-assessment-create.ts

```diff
@@ -0,0 +1,115 @@
+import dayjs from "dayjs";
+import duration from "dayjs/plugin/duration";
+import { chunk } from "lodash";
+import { Patient } from "@metriport/core/domain/patient";
+import { sleep } from "@metriport/shared";
+import { DocumentQueryProgress } from "@metriport/core/domain/document-query";
+import { createPatient, PatientCreateCmd } from "./create-patient";
+import { matchPatient } from "./get-patient";
+import { updatePatient } from "./update-patient";
+import { getPatientOrFail } from "./get-patient";
+import { queryDocumentsAcrossHIEs } from "../document/document-query";
+import { getConsolidated, ConsolidatedData } from "./consolidated-get";
+
+dayjs.extend(duration);
+
+const delayTime = dayjs.duration(30, "seconds").asMilliseconds();
+const dqDrBuffer = dayjs.duration(10, "minutes").asMilliseconds();
+
+export async function createCoverageAssessments({
+  cxId,
+  facilityId,
+  patientCreates,
+}: {
+  cxId: string;
+  facilityId: string;
+  patientCreates: PatientCreateCmd[];
+}): Promise<void> {
+  const patients: Patient[] = [];
+  const pdChunkSize = 50;
+  const pdChunks = chunk(patientCreates, pdChunkSize);
+  for (const chunk of pdChunks) {
+    await sleep(delayTime);
+    const createPatients: Promise<Patient>[] = [];
+    for (const patient of chunk) {
+      createPatients.push(createOrUpdatePatient(patient));
+    }
+    const patientPromises = await Promise.allSettled(createPatients);
+    patientPromises.map(promise => {
+      if (promise.status == "fulfilled") patients.push(promise.value);
+    });
+  }
+  const patientIds = patients.map(patient => patient.id);
+  const dqChunkSize = 10;
+  const dQChunks = chunk(patientIds, dqChunkSize);
+  for (const chunk of dQChunks) {
+    await sleep(delayTime);
+    const docQueries: Promise<DocumentQueryProgress>[] = [];
+    for (const patientId of chunk) {
+      docQueries.push(
+        queryDocumentsAcrossHIEs({
+          cxId,
+          patientId,
+          facilityId,
+        })
+      );
+    }
+    await Promise.allSettled(docQueries);
+  }
+  await sleep(dqDrBuffer);
+  const pollingAttempts = 20;
+  const remainingPatientIds = [...patientIds];
+  for (let i = 0; i < pollingAttempts; i++) {
+    await sleep(delayTime);
+    const drDone: Promise<string | undefined>[] = [];
+    for (const patientId of patientIds) {
+      drDone.push(pollDrDone(patientId, cxId));
+    }
+    const drDonePromises = await Promise.allSettled(drDone);
```

Comment:
I'd name it the other way around, the promises are the ones that we pass to the `.allSettled`, and the result is the actual data :)

Reply:
Done.